marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/flow/job-query-lib.sjs

```diff
@@ -55,10 +57,61 @@ function findStepResponses(query) {
   response["start"] = start;
   response["pageLength"] = pageLength;
   response["results"] = jobsDataResults.toObject();
+  response["facets"] = computeFacets(whereClause);
 
   return response;
 }
 
+function buildWhereClause(selectedFacets) {
+  if(!selectedFacets || !Object.keys(selectedFacets).length) {
+    return "";
+  }
+
+  let whereClause = "WHERE";
+  const keys = Object.keys(selectedFacets);
+  for(let i=0; i<keys.length; i++) {
+    if(keys[i] === 'startTime') {
+      const betweenCondition = selectedFacets[keys[i]].map(element => "'".concat(element).concat("'"));
+      whereClause = whereClause + " " + "Job.StepResponse.stepStartTime" + " BETWEEN " + betweenCondition[0] + "AND " + betweenCondition[1];
+    } else {
+      const inCondition = selectedFacets[keys[i]].map(element => "'".concat(element).concat("'")).join();
+      whereClause = whereClause + " " + "Job.StepResponse.".concat(keys[i]) + " IN (" + inCondition + ")";
+    }
+
+    if(i !== keys.length-1) {
+      whereClause = whereClause.concat(" ").concat("AND");
+    }
+  }
+  return whereClause;
+}
+
+function computeFacets(whereClause) {
+  const queries = buildFacetQueries(whereClause);
+  const facets = {};
+
+  Object.keys(queries).forEach(column => {
+    const query = queries[column];
+    const results = xdmp.sql(query, ["map", "optimize=0"]).toObject();
```

Comment:
Check with Erik on this - ML 10.0-6 has support for facets in the Optic API. We may not be able to make 10.0-6 a requirement for this feature though. But, he may have advice on whether this is the most efficient approach to take.